internal/events/event.go

```diff
@@ -314,14 +317,30 @@ func (e Events) queryEvents(ctx context.Context, selectedCategories []string, cl
 	}
 
 	if result == nil { //non supported option
-		return make([]*common.Event, 0), nil
+		return make([]*common.Event, 0), swag.Int64(0), nil
 	}
 
 	var events []*common.Event
-	return events, WithIDs(result).Find(&events).Error
+
+	result = WithIDs(result)
+	totalRecords := result.Find(&events).RowsAffected
```

Comment:
How is this working under the bonnet? Are we actually executing a full unlimited query, and then again another one with limit below?